rasa_sdk/knowledge_base/actions.py

```diff
@@ -125,20 +125,35 @@ async def run(
         object_type = tracker.get_slot(SLOT_OBJECT_TYPE)
         last_object_type = tracker.get_slot(SLOT_LAST_OBJECT_TYPE)
         attribute = tracker.get_slot(SLOT_ATTRIBUTE)
+        has_mention = tracker.get_slot(SLOT_MENTION) is not None
 
         new_request = object_type != last_object_type
 
         if not object_type:
-            # object type always needs to be set as this is needed to query the
-            # knowledge base
-            dispatcher.utter_message(response="utter_ask_rephrase")
-            return []
+            # sets the object type dynamically from entities if object_type is not found
+            # in user query
+            object_types = await utils.call_potential_coroutine(
+                self.knowledge_base.get_object_types()
+            )
+            object_type_dynamic = await get_object_type_dynamic(tracker, object_types)
 
-        if not attribute or new_request:
-            return await self._query_objects(dispatcher, object_type, tracker)
-        elif attribute:
+        if object_type:
```

Comment:
I would actually assign this to `object_type` then I can remove the `if` check in line 140.

Reply:
We made the following changes:
- In the `if not object_type` block we are assigning the extracted object type to `object_type` variable
- Due to this change, we are checking for `if object_type and attribute` and `elif (object_type and not attribute) or (object_type and new_request)`